lib/src/metta/runner/builtin_mods/json.rs

```diff
@@ -0,0 +1,223 @@
+use crate::metta::runner::bool::*;
+use crate::metta::runner::number::Number;
+use crate::metta::types::{get_atom_types, get_meta_type};
+
+pub static JSON_METTA: &'static str = include_str!("json.metta");
+
+#[derive(Debug)]
+pub(crate) struct JsonModLoader;
+
+impl ModuleLoader for JsonModLoader {
+    fn load(&self, context: &mut RunContext) -> Result<(), String> {
+        // Initialize module's space
+        let space = GroundingSpace::new();
+        context.init_self_module(space.into(), None);
+
+        // Load module's tokens
+        let _ = self.load_tokens(context.module(), context.metta.clone())?;
+
+        // Parse MeTTa code of the module
+        let parser = SExprParser::new(JSON_METTA);
+        context.push_parser(Box::new(parser));
+
+        Ok(())
+    }
+
+    fn load_tokens(&self, target: &MettaMod, _metta: Metta) -> Result<(), String> {
+        let mut tref = target.tokenizer().borrow_mut();
+
+        tref.register_function(GroundedFunctionAtom::new(
+            r"json-encode".into(),
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_ATOM, ATOM_TYPE_STRING]),
+            json_encode));
+
+        tref.register_function(GroundedFunctionAtom::new(
+            r"json-decode".into(),
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_STRING, ATOM_TYPE_SPACE]),
+            json_decode));
+
+        Ok(())
+    }
+}
+
+fn encode_list(input: Vec<Atom>) -> Result<String, ExecError> {
+    let mut res_str = "".to_string();
+    for atom in input {
+        match encode_atom(&atom) {
+            Ok(encoded) => {res_str = res_str + ", " + &encoded;},
+            Err(err) => return Err(err),
+        }
+    }
+    res_str.remove(0);
+    res_str.remove(0);
+    Ok(format!("[{}]", res_str))
+}
+
+fn encode_dictspace(input: &Atom) -> Result<String, ExecError> {
+    let space = Atom::as_gnd::<DynSpace>(input).unwrap();
+    let result = space.borrow().subst(&Atom::expr([Atom::var("k"), Atom::var("v")]),
+                                      &Atom::expr([Atom::var("k"), Atom::var("v")]));
+    if result.len() == 0 {
+        return Err(ExecError::from("Input space must be a dictionary space which contains (key value) pairs. See dict-space function"));
+    }
+    let mut res_str = "".to_string();
+    for res_atom in result {
+        let res_expr = TryInto::<&ExpressionAtom>::try_into(&res_atom)?;
+        let key = match Str::from_atom(res_expr.children().get(0).unwrap()).ok_or_else(|| ExecError::from("Key must be a string")){
+            Ok(str) => str.to_string(),
+            Err(err) => return Err(err),
+        };
+        let value = res_expr.children().get(1).unwrap();
+        let value_str = encode_atom(value)?;
+        res_str = res_str + ", " + &key + ": " + &value_str;
+    }
+    res_str.remove(0);
+    res_str.remove(0);
+    res_str = format!("{{{}}}", res_str);
+    Ok(res_str)
+}
+
+fn encode_atom(input: &Atom) -> Result<String, ExecError> {
+    let metatype = atom_to_string(&get_meta_type(&input));
```

Comment:
It is unnecessary to convert the meta-type into a string. If you need to check if it is expression you even don't need a meta-type. You can just use `TryInto::<&ExpressionAtom>` results to check whether it is expression or not.